File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int len = Integer.MAX_VALUE;;
        int sum = 0;
        int n = nums.length;
        int left = 0;
        for (int right = 0; right < n; right++) {
            sum += nums[right];
            while (sum >= target) {
                len = Math.min(len, right - left + 1);
                sum -= nums[left++];
            }
        }
        return len == Integer.MAX_VALUE ? 0 : len;

    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        l, total = 0,0
        res = inf
        for r in range(len(nums)):
            total += nums[r]
            while total >= target:
                res = min(res, r - l + 1)
                total -= nums[l]
                l += 1
            
            

        return 0 if res == inf else res 